src/main/java/com/study/realworld/domain/user/presentation/UserApi.java

```diff
@@ -1,45 +1,53 @@
 
-import com.study.realworld.domain.user.application.UserFindService;
 import com.study.realworld.domain.auth.dto.ResponseToken;
 import com.study.realworld.domain.auth.infrastructure.TokenProvider;
 import com.study.realworld.domain.user.application.UserJoinService;
+import com.study.realworld.domain.user.application.UserUpdateService;
 import com.study.realworld.domain.user.domain.User;
 import com.study.realworld.domain.user.dto.UserJoinRequest;
 import com.study.realworld.domain.user.dto.UserResponse;
-import org.slf4j.Logger;
+import com.study.realworld.domain.user.dto.UserUpdateRequest;
 import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
 import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RestController;
 
 import javax.validation.Valid;
 
-import static org.slf4j.LoggerFactory.getLogger;
-
 @RestController
-public class UserRestController {
-
-    private static final String JOIN_REQUEST_LOG_MESSAGE = "join request: {}";
-
-    private final Logger log = getLogger(getClass()); // 유틸성이라 개행
+public class UserApi {
 
     private final UserJoinService userJoinService;
+    private final UserUpdateService userUpdateService;
     private final TokenProvider tokenProvider;
 
-    public UserRestController(final UserJoinService userJoinService,
-                              final TokenProvider tokenProvider) {
+    public UserApi(final UserJoinService userJoinService,
+                   final UserUpdateService userUpdateService,
+                   final TokenProvider tokenProvider) {
         this.userJoinService = userJoinService;
+        this.userUpdateService = userUpdateService;
         this.tokenProvider = tokenProvider;
     }
 
     @PostMapping("/api/users")
     public ResponseEntity<UserResponse> join(@Valid @RequestBody final UserJoinRequest userJoinRequest) {
-        log.info(JOIN_REQUEST_LOG_MESSAGE, userJoinRequest.toString());
         final User user = userJoinService.join(userJoinRequest.toUser());
         final ResponseToken responseToken = tokenProvider.createToken(user);
         final UserResponse userResponse = UserResponse.fromUserWithToken(user, responseToken);
         return ResponseEntity.ok().body(userResponse);
     }
 
+
+    @PutMapping("/api/users")
+    public ResponseEntity<UserResponse> update(@Valid @RequestBody final UserUpdateRequest userUpdateRequest,
+                                               @AuthenticationPrincipal final Long principal) {
+        final User user = userUpdateService.update(userUpdateRequest.toEntity(), principal);
+        final ResponseToken responseToken = tokenProvider.createToken(user);
+        final UserResponse userResponse = UserResponse.fromUserWithToken(user, responseToken);
```

Comment:
```suggestion
        final UserResponse userResponse = UserResponse.fromUserAndToken(user, responseToken);
```
개인적으로는 문맥상 And 가 더 나아보일듯 하네요

Reply:
오잉 왜 With 이 들어갔는지 제가 봐도 이해 안가네요